models/user_model.go

```diff
@@ -8,13 +8,22 @@ import (
 	"go.mongodb.org/mongo-driver/mongo"
 )
 
-func FindProfile(username string) (userProfile bson.M, err error) {
+func FindProfile(username string) (resultMap map[string]interface{}, err error) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	var result bson.M
 	err = userCollection.FindOne(ctx, bson.M{"username": username}).Decode(&result)
+	if err != nil {
+		return nil, err
+	}
+
+	delete(result, "_id")
+	delete(result, "createdAt")
+	delete(result, "updatedAt")
+	delete(result, "hashedPassword")
```

Comment:
You should be doing a mongodb projection instead. This is to reduce pulling lots of data from db.